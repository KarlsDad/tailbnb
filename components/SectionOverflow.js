export default ({ title, children }) => {
  return (
    <div className='pl-5 sm:px-5 md:px-10 xl:px-0 xl:px-0 lg:max-w-5.5xl mx-auto py-5 overflow-x-hidden overflow-y-hidden w-full'>
      <h3 className='text-gray-850 text-2xl font-semibold py-3 '>{title}</h3>
      {children}
    </div>
  );
};
