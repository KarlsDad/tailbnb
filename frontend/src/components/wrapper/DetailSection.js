import React from 'react';

export default ({ title, children }) => {
  return (
    <div className='border-b border-gray-300 pt-4 pb-14 w-full'>
      <div className='pb-8'>
        <h3
          style={{ fontFamily: 'airbnb-medium' }}
          className='text-2xl my-4 text-gray-850'>
          {title}
        </h3>
        {children}
      </div>
    </div>
  );
};
