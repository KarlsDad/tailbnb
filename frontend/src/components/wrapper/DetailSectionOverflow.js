import React from 'react';

export default ({ title, phrase, children }) => {
  function renderPhrase() {
    if (phrase) {
      return (
        <>
          <h3
            style={{ fontFamily: 'airbnb-medium' }}
            className='scrollable text-gray-850 text-2xl pt-3 w-5/6'>
            {title}
          </h3>
          <p
            style={{ fontFamily: 'airbnb-book' }}
            className='scrollable text-md mb-5 text-gray-850 pb-3'>
            {phrase}
          </p>
        </>
      );
    } else {
      return (
        <>
          <h3
            style={{ fontFamily: 'airbnb-medium' }}
            className='text-2xl my-4 text-gray-850'>
            {title}
          </h3>
        </>
      );
    }
  }
  return (
    <div className='border-b border-gray-300 pt-4 pb-14 w-full mx-auto overflow-x-hidden overflow-y-hidden'>
      <div className='pb-8'>
        <h3
          style={{ fontFamily: 'airbnb-medium' }}
          className='text-2xl my-4 text-gray-850'>
          {title}
        </h3>

        {children}
      </div>
    </div>
  );
};
