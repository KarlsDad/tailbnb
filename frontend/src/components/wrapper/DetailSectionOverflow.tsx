import * as React from 'react';

interface Prop {
  title: string | null;
  phrase?: string | undefined;
  children: React.ReactNode;
}

export const DetailSectionOverflow: React.FC<Prop> = ({
  title,
  phrase,
  children
}) => {
  function renderPhrase(phrase: string | undefined) {
    if (phrase) {
      return (
        <>
          <h3
            style={{ fontFamily: 'airbnb-medium' }}
            className='scrollable text-gray-850 text-2xl pt-3 w-5/6'>
            {title}
          </h3>
          <p
            style={{ fontFamily: 'airbnb-book' }}
            className='scrollable text-md mb-5 text-gray-850 pb-3'>
            {phrase}
          </p>
        </>
      );
    } else {
      return (
        <>
          <h3
            style={{ fontFamily: 'airbnb-medium' }}
            className='text-2xl my-4 text-gray-850'>
            {title}
          </h3>
        </>
      );
    }
  }
  return (
    <div className='border-b border-gray-300 pt-4 pb-14 w-full mx-auto overflow-x-hidden overflow-y-hidden'>
      <div className='pb-8'>
        {renderPhrase(phrase)}
        {children}
      </div>
    </div>
  );
};
