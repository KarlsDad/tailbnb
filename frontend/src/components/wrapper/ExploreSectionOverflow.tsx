import * as React from 'react';

interface Prop {
  title: string;
  phrase: string;
  children: React.ReactNode;
}

export const ExploreSectionOverflow: React.FC<Prop> = ({
  title,
  phrase,
  children
}) => {
  function renderPhrase(phrase: string | undefined) {
    if (phrase) {
      return (
        <>
          <h3
            style={{ fontFamily: 'airbnb-medium' }}
            className='scrollable text-gray-850 text-2xl pt-3 w-5/6'>
            {title}
          </h3>
          <p
            style={{ fontFamily: 'airbnb-book' }}
            className='scrollable text-md mb-5 text-gray-850 pb-3'>
            {phrase}
          </p>
        </>
      );
    } else {
      return (
        <>
          <h3 className='scrollable text-gray-850 text-2xl font-semibold py-3 w-5/6'>
            {title}
          </h3>
        </>
      );
    }
  }
  return (
    <div className='md:px-0 max-w-90% mx-auto py-5 overflow-x-hidden overflow-y-hidden'>
      {renderPhrase(phrase)}
      {children}
    </div>
  );
};
