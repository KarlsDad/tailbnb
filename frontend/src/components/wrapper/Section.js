import React from 'react';

export default ({ title, phrase, children }) => {
  const renderContent = (phrase) => {
    if (phrase) {
      return (
        <>
          <h3
            style={{ fontFamily: 'airbnb-medium' }}
            className='text-2xl text-gray-850 pt-3'>
            {title}
          </h3>
          <p
            style={{ fontFamily: 'airbnb-book' }}
            className='text-md mb-5 text-gray-850 pb-3'>
            {phrase}
          </p>
        </>
      );
    } else {
      return (
        <>
          <h3
            style={{ fontFamily: 'airbnb-bold' }}
            className='text-gray-750 text-2xl pt-3 pb-5'>
            {title}
          </h3>
        </>
      );
    }
  };
  return (
    <div className='px-5 md:px-10 lg:max-w-5.5xl xl:px-0 mx-auto py-5 w-full'>
      {renderContent(phrase)}
      {children}
    </div>
  );
};
