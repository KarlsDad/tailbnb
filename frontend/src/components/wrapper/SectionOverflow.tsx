import * as React from 'react';

interface Prop {
  title: string;
  phrase?: string | undefined;
  children: React.ReactNode;
}

export const SectionOverflow: React.FC<Prop> = ({
  title,
  phrase,
  children
}) => {
  function renderPhrase(phrase: string | undefined) {
    if (phrase) {
      return (
        <>
          <h3
            style={{ fontFamily: 'airbnb-medium' }}
            className='scrollable text-gray-850 text-2xl pt-3 w-5/6'>
            {title}
          </h3>
          <p
            style={{ fontFamily: 'airbnb-book' }}
            className='scrollable text-md mb-5 text-gray-850 pb-3'>
            {phrase}
          </p>
        </>
      );
    } else {
      return (
        <>
          <h3
            style={{ fontFamily: 'airbnb-medium' }}
            className='scrollable text-gray-850 text-2xl py-3 w-5/6'>
            {title}
          </h3>
        </>
      );
    }
  }
  return (
    <div className='px-6 md:px-8 lg:px-10 xl:px-20 xl:max-w-layout mx-auto py-5 overflow-x-hidden overflow-y-hidden'>
      {renderPhrase(phrase)}
      {children}
    </div>
  );
};
