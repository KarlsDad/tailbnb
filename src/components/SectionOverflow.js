import React from 'react';

export default ({ title, phrase, children }) => {
  function renderPhrase() {
    if (phrase) {
      return (
        <>
          <h3 className='scrollable text-gray-850 text-2xl font-semibold py-3 '>
            {title}
          </h3>
          <p className='scrollable text-sm md:text-md mb-5 text-gray-850 pb-3'>
            {phrase}
          </p>
        </>
      );
    } else {
      return (
        <>
          <h3 className='scrollable text-gray-850 text-2xl font-semibold py-3 '>
            {title}
          </h3>
        </>
      );
    }
  }
  return (
    <div className='sm:px-5 md:px-10 xl:px-0 lg:max-w-5.5xl mx-auto py-5 overflow-x-hidden overflow-y-hidden w-full'>
      {renderPhrase()}
      {children}
    </div>
  );
};
