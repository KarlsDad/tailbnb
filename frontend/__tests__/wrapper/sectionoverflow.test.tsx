import * as React from 'react';
import { shallow } from 'enzyme';

// Component
import { SectionOverflow } from '../../src/components/wrapper/SectionOverflow';

describe('SectionOverflow', () => {
  it('should render correctly in "debug" mode', () => {
    const info = {
      title: 'test title',
      phrase: 'test phrase',
      children: null
    };
    const component = shallow(<SectionOverflow {...info} />);
    expect(component).toMatchSnapshot();
  });
});
