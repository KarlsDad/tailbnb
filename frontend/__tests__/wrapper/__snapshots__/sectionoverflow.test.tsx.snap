// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionOverflow should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
